numpyro/distributions/directional.py

```diff
@@ -19,6 +19,143 @@
+            phi_conc = numpyro.sample('phi_conc', Beta(2., 2.))
+            psi_conc = numpyro.sample('psi_conc', Beta(2., 2.))
+            corr_scale = numpyro.sample('corr_scale', Beta(2., 5.))
+
+            # SS prior
+            skew_phi = numpyro.sample('skew_phi', Uniform(-1., 1.))
+            psi_bound = 1 - skew_phi.abs()
+            skew_psi = numpyro.sample('skew_psi', Uniform(-1., 1.))
+            skewness = torch.stack((skew_phi, psi_bound * skew_psi), dim=-1)
+            with numpyro.plate('obs_plate'):
+                sine = SineBivariateVonMises(phi_loc=phi_loc, psi_loc=psi_loc,
+                                            phi_concentration=1000 * phi_conc,
+                                            psi_concentration=1000 * psi_conc,
+                                            weighted_correlation=corr_scale)
+                return numpyro.sample('phi_psi', SineSkewed(sine, skewness), obs=obs)
+
+    To ensure the skewing does not alter the normalization constant of the (Sine Bivaraite von Mises) base
+    distribution the skewness parameters are constraint. The constraint requires the sum of the absolute values of
+    skewness to be less than or equal to one.
+    So for the above snippet it must hold that::
+
+        skew_phi.abs()+skew_psi.abs() <= 1
+
+    We handle this in the prior by computing psi_bound and use it to scale skew_psi.
+    We do **not** use psi_bound as::
+
+        skew_psi = pyro.sample('skew_psi', Uniform(-psi_bound, psi_bound))
+
+    as it would make the support for the Uniform distribution dynamic.
+    In the context of :class:`~pyro.infer.SVI`, this distribution can freely be used as a likelihood. But, when used as
+    latent variables it will lead to slow inference for 2 and higher dim toruses. This is because the base_dist
+    cannot be reparameterized.
+
+    .. note:: An event in the base distribution must be on a d-torus, so the event_shape must be (d,).
+
+    .. note:: For the skewness parameter, it must hold that the sum of the absolute value of its weights for an event
+        must be less than or equal to one. See eq. 2.1 in [1].
+
+    ** References: **
+        1. Sine-skewed toroidal distributions and their application in protein bioinformatics
+            Ameijeiras-Alonso, J., Ley, C. (2019)
+
+    :param torch.distributions.Distribution base_dist: base density on a d-dimensional torus. Supported base
+        distributions include: 1D :class:`~numpyro.distributions.VonMises`,
+        :class:`~numnumpyro.distributions.SineBivariateVonMises`, 1D :class:`~numpyro.distributions.ProjectedNormal`,
+        and :class:`~numpyro.distributions.Uniform` (-pi, pi).
+    :param torch.tensor skewness: skewness of the distribution.
+    """
+
+    arg_constraints = {
+        "skewness": constraints.independent(constraints.interval(-1.0, 1.0), 1)
+    }
+
+    support = constraints.independent(constraints.real, 1)
```

Comment:
similar to the other PR, add a TODO for further discussion

Reply:
done C: